### PR TITLE
Add AV production devices to hwdb

### PR DESCRIPTION
**Pull request by GeorgesStavracas**
Builds on top of https://github.com/systemd/systemd/pull/20561

This adds support for multimedia devices, including everything in https://github.com/systemd/systemd/pull/20561, but also Elgato Stream Deck devices.

I did not apply any of the corrections or comments from https://github.com/systemd/systemd/pull/20561, but will be happy to take it over and push this to the finish line if @Be-ing agrees.

CC @hadess

### CODE REVIEW

**Comment by hadess** [comment]
> what is "multimedia" device? back in my day that was the combination of a cdrom drive with a soundblaster card. I doubt that's what's meant here...
> 
> This needs better explanatory comments in the commit msg/PR, as boring people like me have no clue what "multimedia" and "steam deck" is supposed to be?

I think "AV production" equipment is a better term.

File: hwdb.d/meson.build [hunk 1]
```diff
@@ -16,7 +16,8 @@ hwdb_files_notest = files(
         '20-acpi-vendor.hwdb',
         '20-OUI.hwdb',
         '20-net-ifname.hwdb',
-        '20-vmbus-class.hwdb')
+        '20-vmbus-class.hwdb',
+        '70-multimedia.hwdb')
```
Comment (hadess) [inline hwdb.d/meson.build]: This obviously needs to be removed. The `props` variable in `hwdb.d/parse_hwdb.py` was modified to cover this case. Why doesn't it work? What's the error message?

File: hwdb.d/70-multimedia.hwdb [hunk 2]
```diff
@@ -22,6 +22,29 @@ usb:v2982p1967*
 usb:v1157p300*
  ID_MULTIMEDIA_CONTROLLER=1
 
+################
+# Elgato
+################
+# Stream Deck Original (gen 1)
+usb:v0FD9p0060*
+ ID_MULTIMEDIA_CONTROLLER=1
```
Comment (hadess) [inline hwdb.d/70-multimedia.hwdb]: Add a link to the "Stream Deck" product page on Elgato's website, or a Wikipedia link, in the commit message.

File: hwdb.d/70-multimedia.hwdb [hunk 3]
```diff
@@ -1,6 +1,6 @@
 # This file is part of systemd.
 #
-# Database for DJ and music production controllers that should be accessible to the seat owner.
+# Database for multimedia production controllers that should be accessible to the seat owner.
```
Comment (hadess) [inline hwdb.d/70-multimedia.hwdb]: As mentioned in a comment, I think that `ID_AV_PRODUCTION_CONTROLLER` is a good match for the devices we're talking about.

The comment here should be longer though, explaining the types of devices we're dealing with, maybe with authoritative links to we're talking about.

Comment (hadess) [reply hwdb.d/70-multimedia.hwdb]: This commit should be merged into the previous one, keep the authorship and add yourself as a co-author:
https://docs.kernel.org/process/submitting-patches.html#when-to-use-acked-by-cc-and-co-developed-by

File: hwdb.d/meson.build (hunk 1 above)
Comment (GeorgesStavracas) [reply hwdb.d/meson.build]: It fails with

```
./70-av-production.hwdb: 29 match groups, 29 matches, 29 properties
Pattern 'v1157p300*' is invalid: Expected '*', found 'p300'  (at char 5), (line:1, col:6)
```

I don't understand why it fails there. There are other files with the same pattern there

Comment (GeorgesStavracas) [reply hwdb.d/meson.build]: Perhaps the product id should be `0300`, actually...

Comment (GeorgesStavracas) [reply hwdb.d/meson.build]: Done

File: hwdb.d/70-multimedia.hwdb (hunk 2 above)
Comment (GeorgesStavracas) [reply hwdb.d/70-multimedia.hwdb]: Added a link to https://www.elgato.com/en/stream-deck in the commit message

File: hwdb.d/70-multimedia.hwdb (hunk 3 above)
Comment (GeorgesStavracas) [reply hwdb.d/70-multimedia.hwdb]: Changed to `ID_AV_PRODUCTION_CONTROLLER`, added myself as co-author in the commit message while retaining original authorship in the commit itself.

Comment (GeorgesStavracas) [reply hwdb.d/70-multimedia.hwdb]: ... and updated header comment in the 70-av-production.hwdb file

File: hwdb.d/meson.build (hunk 1 above)
Comment (GeorgesStavracas) [reply hwdb.d/meson.build]: A quick Google search shows that most devices of this vendor do start with `03`, e.g. `0300` (this case), `0310`, etc.

**Comment by GeorgesStavracas** [comment]
Ah, sorry, I just noticed @poettering requested better commit messages. Let's see what I can come up with...

**Comment by GeorgesStavracas** [comment]
Done, hopefully this should be more explanatory.

@poettering for some additional context, I've been writing an app to control Stream Deck devices, and noticed they couldn't be read without root access. @hadess oriented me to https://github.com/systemd/systemd/pull/20561, and generally suggested the approach I've taken here (thanks Bastien!).

**Comment by Be-ing** [comment]
I liked the term "multimedia", but "AV" works too I guess :shrug:

**Comment by hadess** [comment]
Looks good to me.

**Comment by hadess** [comment]
> I liked the term "multimedia", but "AV" works too I guess shrug

This is what multimedia means for old farts like @poettering and I:
image

**Comment by GeorgesStavracas** [comment]
No idea why this particular CentOS image failed CI. Nothing obvious comes to my mind by looking at the logs. If anyone knows if this was related or not to this PR, please let me know.

**Comment by mrc0mmand** [comment]
> No idea why this particular CentOS image failed CI. Nothing obvious comes to my mind by looking at the logs. If anyone knows if this was related or not to this PR, please let me know.

It definitely unrelated. I restarted the job (and added it to my to-investigate list).

**Comment by Be-ing** [comment]
I'm okay with merging this if @GeorgesStavracas confirms that he has tested it locally with the devices he has added.

**Comment by GeorgesStavracas** [comment]
Installing this new hwdb file, running the appropriate commands, and restarting my computer (just in case), and I do get user access on my Stream Deck device. This *seems* to be working here.

**Review by bluca — APPROVED** [review]

**Comment by GeorgesStavracas** [comment]
Thank you!

**Comment by poettering** [comment]
Also see #21743.

**Pull request merged**